src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -194,5 +215,17 @@ internal void SetInitializedWorkerChannel(string initializedRuntime, ILanguageWo
                 }
             }
         }
+
+        internal void SetExceptionOnInitializedWorkerChannel(string initializedRuntime, ILanguageWorkerChannel initializedLanguageWorkerChannel, Exception exception)
+        {
+            _logger.LogDebug("Failed to initialize webhost language worker channel for runtime: {language}. workerId:{id}", initializedRuntime, initializedLanguageWorkerChannel.Id);
+            if (_workerChannels.TryGetValue(initializedRuntime, out Dictionary<string, TaskCompletionSource<ILanguageWorkerChannel>> channel))
+            {
+                if (channel.TryGetValue(initializedLanguageWorkerChannel.Id, out TaskCompletionSource<ILanguageWorkerChannel> value))
+                {
+                    value.SetException(exception);
```

Comment:
I think that the logic for removing this faulted channel from _workerChannels is in FunctionDispatcher's handling of WorkerError, but just want to double check!

Reply:
Nevermind - it looks like FunctionDispatcher maintains correct order - resolving!